src/wui/buildingwindow.cc

```diff
@@ -500,6 +518,27 @@ void BuildingWindow::act_start_stop() {
 	}
 }
 
+/**
+===============
+Callback for toggling the production site's infinite production mode
+===============
+*/
+void BuildingWindow::act_produce_infinite() {
+	Widelands::Building* building = building_.get(parent_->egbase());
+	if (building == nullptr) {
+		return;
+	}
+
+	if (building->descr().type() >= Widelands::MapObjectType::PRODUCTIONSITE) {
+		if (game_ != nullptr) {
+			game_->send_player_toggle_infinite_production(*building);
+		} else {
+			Widelands::ProductionSite& ps = dynamic_cast<Widelands::ProductionSite&>(*building);
```

Comment:
when do we not have a game and set this? or what is the purpose here and why here and not in the start Stop for example

Reply:
This is for the future scenario editor where all properties of map objects need to be changed directly instead of sending a playercommand.